tiatoolbox/cli.py

```
@@ -323,11 +327,111 @@ def stainnorm(source_input, target_input, method, stain_matrix, output_dir, file
+)
+@click.option(
+    "--kernel_size",
+    type=int,
+    nargs=2,
+    help="kernel size for morphological dilation, default=1, 1",
+)
+@click.option(
+    "--mode",
+    default="show",
+    help="'show' to display tissue mask or 'save' to save at the output path"
+    ", default=show",
+)
+@click.option(
+    "--file_types",
+    help="file types to capture from directory, "
+    "default='*.svs, *.ndpi, *.jp2, *.png', '*.jpg', '*.tif', '*.tiff'",
+    default="*.svs, *.ndpi, *.jp2, *.png, *.jpg, *.tif, *.tiff",
+)
+def tissue_mask(
+    wsi_input, output_path, method, resolution, units, kernel_size, mode, file_types
+):
+    """Generate tissue mask for a WSI."""
+
+    file_types = tuple(file_types.split(", "))
+    output_path = pathlib.Path(output_path)
+    if os.path.isdir(wsi_input):
+        files_all = utils.misc.grab_files_from_dir(
+            input_path=wsi_input, file_types=file_types
+        )
+    elif os.path.isfile(wsi_input):
+        files_all = [
+            wsi_input,
+        ]
+    else:
+        raise FileNotFoundError
+
+    if mode == "save" and not output_path.is_dir():
+        os.makedirs(output_path)
+
+    if method == "Otsu":
+        masker = tissuemask.OtsuTissueMasker()
+    elif method == "Morphological":
+        if not kernel_size:
+            if units == "mpp":
+                masker = tissuemask.MorphologicalMasker(mpp=resolution)
+            elif units == "power":
+                masker = tissuemask.MorphologicalMasker(power=resolution)
+            else:
+                raise MethodNotSupported(
+                    "Specified units not supported for tissue masking."
+                )
+        else:
+            masker = tissuemask.MorphologicalMasker(kernel_size=kernel_size)
+    else:
+        raise MethodNotSupported
+
+    for curr_file in files_all:
+        wsi = wsicore.wsireader.get_wsireader(input_img=curr_file)
+        wsi_thumb = wsi.slide_thumbnail(resolution=1.25, units="power")
```

@John-P , don't you think that here the input `resolution` and `unit` from the user should be used instead of default values (resolution=1.25, units="power") to achieve optimal concordance between `masker` and `slide_thumbnail`?

---

I think default thumbnail size is at 1.25 power that's why default for tissue mask seems good to me.